examples/tv-casting-app/darwin/TvCasting/TvCasting/MCContentLauncherLaunchURLExampleViewModel.swift

```diff
@@ -0,0 +1,83 @@
+/**
+ *
+ *    Copyright (c) 2020-2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+
+import Foundation
+import os.log
+
+class MCContentLauncherLaunchURLExampleViewModel: ObservableObject {
+    let Log = Logger(subsystem: "com.matter.casting",
+                     category: "MCContentLauncherLaunchURLExampleViewModel")
+
+    @Published var status: String?;
+
+    // VendorId of the MCEndpoint on the MCCastingPlayer that the MCCastingApp desires to interact with after connection
+    let sampleEndpointVid: Int = 65521
+
+    func invokeCommand(castingPlayer: MCCastingPlayer, contentUrl: String, displayString: String)
+    {
+        // select the MCEndpoint on the MCCastingPlayer to invoke the command on
+        if let endpoint: MCEndpoint = castingPlayer.endpoints().filter({ $0.vendorId().intValue == sampleEndpointVid}).first
+        {
+            // get ContentLauncher cluster from the endpoint
+            if(endpoint.hasCluster(MCEndpointClusterTypeContentLauncher))
+            {
+                let contentLaunchercluster: MCContentLauncherCluster = endpoint.cluster(for: MCEndpointClusterTypeContentLauncher) as! MCContentLauncherCluster
+
+                // get the launchURLCommand from the contentLauncherCluster
+                let launchURLCommand: MCContentLauncherClusterLaunchURLCommand? = contentLaunchercluster.launchURLCommand()
+                if(launchURLCommand != nil)
+                {
+                    // create the LaunchURL request
+                    let request: MCContentLauncherClusterLaunchURLRequest = MCContentLauncherClusterLaunchURLRequest()
+                    request.contentURL = contentUrl
+                    request.displayString = displayString
+                    
+                    // call invoke on launchURLCommand while passing in a completion block
+                    launchURLCommand!.invoke(request, context: nil, completion: { context, err, response in
```

Comment:
there's like 4 nested if statements here, any way we could make this easier to read at all?

Reply:
Done. Made some early returns to reduce the if/else nesting.